src/std_extensions/arithmetic/int_ops.rs

```diff
@@ -40,100 +41,145 @@ fn inarrow_sig(arg_values: &[TypeArg]) -> Result<FunctionType, SignatureError> {
-    ))
+fn btoi_sig(
+    int_type_var: Type,
+    temp_reg: &ExtensionRegistry,
+) -> Result<PolyFuncType, SignatureError> {
+    int_polytype(1, type_row![BOOL_T], vec![int_type_var], temp_reg)
 }
 
-fn ibinop_sig(arg_values: &[TypeArg]) -> Result<FunctionType, SignatureError> {
-    let [arg] = collect_array(arg_values);
-    Ok(FunctionType::new(
-        vec![int_type(arg.clone()); 2],
-        vec![int_type(arg.clone())],
-    ))
+fn icmp_sig(
+    int_type_var: Type,
+    temp_reg: &ExtensionRegistry,
+) -> Result<PolyFuncType, SignatureError> {
+    int_polytype(1, vec![int_type_var; 2], type_row![BOOL_T], temp_reg)
 }
 
-fn iunop_sig(arg_values: &[TypeArg]) -> Result<FunctionType, SignatureError> {
-    let [arg] = collect_array(arg_values);
-    Ok(FunctionType::new(
-        vec![int_type(arg.clone())],
-        vec![int_type(arg.clone())],
-    ))
+fn ibinop_sig(
+    int_type_var: Type,
+    temp_reg: &ExtensionRegistry,
+) -> Result<PolyFuncType, SignatureError> {
+    int_polytype(
+        1,
+        vec![int_type_var.clone(); 2],
+        vec![int_type_var.clone()],
+        temp_reg,
+    )
 }
 
-fn idivmod_checked_sig(arg_values: &[TypeArg]) -> Result<FunctionType, SignatureError> {
-    let [arg0, arg1] = collect_array(arg_values);
-    let intpair: TypeRow = vec![int_type(arg0.clone()), int_type(arg1.clone())].into();
-    Ok(FunctionType::new(
+fn iunop_sig(
+    int_type_var: Type,
+    temp_reg: &ExtensionRegistry,
+) -> Result<PolyFuncType, SignatureError> {
+    int_polytype(1, vec![int_type_var.clone()], vec![int_type_var], temp_reg)
+}
+
+fn idivmod_checked_sig(
+    int_type_var_0: Type,
+    int_type_var_1: Type,
+    temp_reg: &ExtensionRegistry,
+) -> Result<PolyFuncType, SignatureError> {
+    let intpair: TypeRow = vec![int_type_var_0, int_type_var_1].into();
+    int_polytype(
+        2,
         intpair.clone(),
         vec![Type::new_sum(vec![Type::new_tuple(intpair), ERROR_TYPE])],
-    ))
+        temp_reg,
+    )
 }
 
-fn idivmod_sig(arg_values: &[TypeArg]) -> Result<FunctionType, SignatureError> {
-    let [arg0, arg1] = collect_array(arg_values);
-    let intpair: TypeRow = vec![int_type(arg0.clone()), int_type(arg1.clone())].into();
-    Ok(FunctionType::new(
-        intpair.clone(),
-        vec![Type::new_tuple(intpair)],
-    ))
+fn idivmod_sig(
+    int_type_var_0: Type,
+    int_type_var_1: Type,
+    temp_reg: &ExtensionRegistry,
+) -> Result<PolyFuncType, SignatureError> {
+    let intpair: TypeRow = vec![int_type_var_0, int_type_var_1].into();
+    int_polytype(2, intpair.clone(), vec![Type::new_tuple(intpair)], temp_reg)
 }
 
-fn idiv_checked_sig(arg_values: &[TypeArg]) -> Result<FunctionType, SignatureError> {
-    let [arg0, arg1] = collect_array(arg_values);
-    Ok(FunctionType::new(
-        vec![int_type(arg0.clone()), int_type(arg1.clone())],
-        vec![Type::new_sum(vec![int_type(arg0.clone()), ERROR_TYPE])],
-    ))
+fn idiv_checked_sig(
+    int_type_var_0: Type,
```

Comment:
Yeah, presumably the only legal values for these two are `int_type_var(0, ...)` and `int_type_var(1, ...)`